# Практическое задание №4 по теме "Непрерывные случайные величины. Функция распределения и функция плотности. Нормальное распределение. Центральная предельная теорема".

In [1]:
import numpy as np
from scipy import stats

from matplotlib import pyplot as plt
plt.style.use('seaborn-whitegrid')
%config InlineBackend.figure_formats = ['svg']

## Задача 1

Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

___Решение___

In [2]:
a = stats.uniform.rvs(loc=200, scale=600, size=100000000)

In [3]:
a_mean = a.mean()
a_var = a.var()
print(f"Среднее значение: {np.round(a_mean, 5)}\nДисперсия: {np.round(a_var, 5)}")

Среднее значение: 499.9889
Дисперсия: 30000.25138


___Ответ:___ Среднее значение: 499.9889, Дисперсия: 30000.25138

---

## Задача 2

О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

___Решение___

In [4]:
right_border = 0.5
left_border = 0.5
k = 3
step = np.power(0.1, k)
n = 1000000
result = 0
for i in range(n):
    right_border += step
    b = np.random.uniform(left_border, right_border, 10000000)
    b_var = b.var(ddof=1)
    if (b_var >= 0.2):
        result = right_border
        break
print(f"Правая граница величины B: {np.round(result, k)}\nДисперсия: {np.round(b_var, k)}")

Правая граница величины B: 2.05
Дисперсия: 0.2


Проверка:

In [5]:
b = np.random.uniform(0.5, 2.05, 10000000)
b.var(ddof=1)

0.20010342392406386

In [6]:
# Среднее значение
np.round(b.mean(), 3)

1.275

___Ответ:___ Правая граница: 2.05, Среднее значение: 1.275

---

## Задача 3

Непрерывная случайная величина X распределена нормально и задана плотностью распределения $f(x)=\frac{1}{4\sqrt{2\pi}}\cdot\frac{e^{-(x+2)^{2}}}{32}$

Найдите:

    а) M(X)
    б) D(X)
    в) std(X) (среднее квадратичное отклонение)

___Решение___

Выражение $f(x)=\frac{1}{4\sqrt{2\pi}}\cdot\frac{e^{-(x+2)^{2}}}{32}$ соответствует нормальному распределнию с параметрами $\mu = -2$ и $\sigma = 4$ 

In [7]:
mu = -2
sigma = 4
X = stats.norm(loc=mu, scale=sigma)

In [8]:
X_mean = X.mean()
X_var = X.var()
X_std = X.std()

In [9]:
X_mean, X_var, X_std

(-2.0, 16.0, 4.0)

___Ответ:___ _а)_ $M(X) = -2$; _б)_ $D(X) = 16$; _в)_ $std(X) = 4$

---

## Задача 4

Рост взрослого населения города X имеет нормальное распределение. Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см. Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:

    а) больше 182 см
    б) больше 190 см
    в) от 166 см до 190 см
    г) от 166 см до 182 см
    д) от 158 см до 190 см
    е) не выше 150 см или не ниже 190 см
    ё) не выше 150 см или не ниже 198 см
    ж) ниже 166 см.


___Решение___

Параметры нормального распределения из условия задания: $\mu = 174$ и $\sigma = 8$ 

In [10]:
mu = 174
sigma = 8
X = stats.norm(loc=mu, scale=sigma)

In [11]:
a_ = 1 - X.cdf(182)
b_ = 1 - X.cdf(190)
v_ = X.cdf(190) - X.cdf(166)
g_ = X.cdf(182) - X.cdf(166)
d_ = X.cdf(190) - X.cdf(158)
e_ = X.cdf(150) + (1 - X.cdf(190))
i_ = X.cdf(150) + (1 - X.cdf(198))
j_ = X.cdf(166)

In [12]:
result = (a_, b_, v_, g_, d_, e_, i_, j_)
result = np.round(result, 5)

In [13]:
result

array([0.15866, 0.02275, 0.81859, 0.68269, 0.9545 , 0.0241 , 0.0027 ,
       0.15866])

___Ответ:___

    а) 0.15866,
    б) 0.02275,
    в) 0.81859,
    г) 0.68269,
    д) 0.9545,
    е) 0.0241,
    ё) 0.0027,
    ж) 0.15866

---

## Задача 5

На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой $M(X)$ = 178 см и $D(X)$ = 25 кв.см?

___Решение___

Параметры нормального распределения из условия задания: $\mu = M(X) = 178$ и $\sigma = \sqrt{D(X)} = 5$ 

In [14]:
value = 190
mx = 178
dx = 25
sigma = np.sqrt(dx)
deviation = (value - mx)/sigma 
deviation

2.4

___Ответ:___ Отклонение на $2.4\sigma$

---